app/samples/phase_detection/src/flight.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+#include <array>
+#include <cstdint>
+#include <f_core/flight/CPhaseController.h>
+
+/**
+ * Flight Events for a fictional flight
+ */
+enum Events : uint8_t { PadReady, Boost, Coast, Noseover, MainChute, GroundHit, CamerasOff, NumEvents };
```

Comment:
Types go brrrrrrr.

In the dnamic memory world we could also have types but then you just like always have to cast them down to uint8 or something before they go into the system.

Alternatively you can do like a list of Phase* or Phase& and subclass Phase but then you cant really refer to phases by name you have to end up referring to them by index and then youre back out of type safety land again